src/sensitivities/linalg/generic.jl

```diff
@@ -128,3 +128,10 @@ function ∇(::typeof(exp), ::Type{Arg{1}}, p, Y, Ȳ, X::AbstractMatrix)
     F = factorize(Uᵀ)
     return real(F \ (Uᵀ * Ȳ / F .* Z) * Uᵀ)
 end
+
+# Ported from autograd, which uses https://mathoverflow.net/questions/25778/analytical-...
+# formula-for-numerical-derivative-of-the-matrix-pseudo-inverse
```

Comment:
Please use the academic reference here as well:
Golub, Gene H., and Victor Pereyra. "The differentiation of pseudo-inverses and nonlinear least squares problems whose variables separate." SIAM Journal on numerical analysis 10.2 (1973): 413-432.